core/drivers/crypto/caam/hal/common/hal_cfg_dt.c

```diff
@@ -20,145 +19,147 @@ static const char *dt_jr_match_table = {
 	"fsl,sec-v4.0-job-ring",
 };
 
-/**
+/*
  * Finds the Job Ring reserved for the Secure Mode in the DTB
  *
  * @fdt         Reference to the Device Tree
  * @status      Status mask flag of the node to found
- * @start_node  [in/out] Node offset to start in DTB, current node found
+ * @find_node   [out] Node offset found
  */
-static paddr_t find_jr_index(void *fdt, int status, int *start_node)
+static paddr_t find_jr_offset(void *fdt, int status, int *find_node)
 {
 	paddr_t jr_offset = 0;
-	int node;
+	int node = -FDT_ERR_NOTFOUND;
 
-	node = fdt_node_offset_by_compatible(fdt, *start_node,
-					     dt_jr_match_table);
+	node = fdt_node_offset_by_compatible(fdt, 0, dt_jr_match_table);
 
 	while (node != -FDT_ERR_NOTFOUND) {
-		HAL_TRACE("Found Job Ring node status 0x%x", node);
-		if (_fdt_get_status(fdt, node) & status) {
+		HAL_TRACE("Found Job Ring node status @%d", node);
+		if (_fdt_get_status(fdt, node) == status) {
 			HAL_TRACE("Found Job Ring node @%d", node);
 			jr_offset = _fdt_reg_base_address(fdt, node);
-			*start_node = node;
+			*find_node = node;
 			break;
 		}
 
 		node = fdt_node_offset_by_compatible(fdt, node,
 						     dt_jr_match_table);
 	}
 
-	HAL_TRACE("JR Base address return 0x%" PRIxPTR, jr_offset);
+	HAL_TRACE("JR Offset return 0x%" PRIxPTR, jr_offset);
 	return jr_offset;
 }
 
 /*
- * Returns the Job Ring configuration to be used by the TEE
+ * Finds the CAAM Controller definition in the DTB.
+ * If found, ensures it reserved for the Secure OS and disabled for the RichOS.
  *
- * @jrcfg   [out] Job Ring configuration
+ * @fdt         Reference to the Device Tree
  */
-enum CAAM_Status caam_hal_cfg_get_conf(struct caam_jrcfg *jrcfg)
+static int find_ctrl(void *fdt)
 {
-	enum CAAM_Status retstatus = CAAM_FAILURE;
-	vaddr_t ctrl_base = 0;
-	paddr_t jr_offset = 0;
-	size_t size = 0;
-
-	void *fdt = NULL;
 	int node = 0;
 
-	fdt = get_dt();
-	if (!fdt) {
-		EMSG("DTB no present\n");
-		goto exit_get_conf;
-	}
-
+	/*
+	 * Node is defined, check if the CAAM Controller is defined. If not
+	 * use HW hard coded value
+	 */
```

Comment:
I think this comment does not belong to this function.

Reply:
function will be removed